docs/DeveloperGuide.adoc

```diff
@@ -111,156 +111,355 @@ image::DeleteSequenceDiagram.png[]
+JSON storage `JsonUtil`, string manipulation `StringUtil` and others.
+//@@author
 
 == Implementation
 
 This section describes some noteworthy details on how certain features are implemented.
 
-// tag::undoredo[]
-=== [Proposed] Undo/Redo feature
-==== Proposed Implementation
+//@@author jiayushe
+// tag::task[]
 
-The undo/redo mechanism is facilitated by `VersionedAddressBook`.
-It extends `AddressBook` with an undo/redo history, stored internally as an `addressBookStateList` and `currentStatePointer`.
-Additionally, it implements the following operations:
+=== Task Management Feature
 
-* `VersionedAddressBook#commit()` -- Saves the current address book state in its history.
-* `VersionedAddressBook#undo()` -- Restores the previous address book state from its history.
-* `VersionedAddressBook#redo()` -- Restores a previously undone address book state from its history.
+As a algorithmic problem management tool, one of the most important features will be manage tasks that have been done
+and are to be done.
 
-These operations are exposed in the `Model` interface as `Model#commitAddressBook()`, `Model#undoAddressBook()` and `Model#redoAddressBook()` respectively.
+This section will describe in details the current implementation and design considerations of
+the task management feature.
 
-Given below is an example usage scenario and how the undo/redo mechanism behaves at each step.
+==== Current Implementation
 
-Step 1. The user launches the application for the first time. The `VersionedAddressBook` will be initialized with the initial address book state, and the `currentStatePointer` pointing to that single address book state.
+The task management feature mainly involves four main behaviours:
 
-image::UndoRedoState0.png[]
+* `AlgoBase#addTask()` - create a new task for a problem and add it to a specified plan.
+* `AlgoBase#deleteTask()` - delete an existing task from a specified plan.
+* `AlgoBase#doneTask()` - mark a task as done in a specified plan.
+* `AlgoBase#undoneTask()` - mark a task as undone in a specified plan.
 
-Step 2. The user executes `delete 5` command to delete the 5th person in the address book. The `delete` command calls `Model#commitAddressBook()`, causing the modified state of the address book after the `delete 5` command executes to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted address book state.
+Given below is an example usage scenario and how the mechanism for adding tasks behaves at each step.
 
-image::UndoRedoState1.png[]
+The following activity diagram summarizes what happens when a user executes the `AddTask` command:
 
-Step 3. The user executes `add n/David ...` to add a new person. The `add` command also calls `Model#commitAddressBook()`, causing another modified address book state to be saved into the `addressBookStateList`.
+image::task/AddTaskCommandActivityDiagram.png[]
```

Comment:
Should the setPlan function in the rightmost activity be creating a new activation bar in itself?